src/test/ui/const-generics/infer/issue-77092.stderr

```diff
@@ -0,0 +1,9 @@
+error[E0282]: type annotations needed
+  --> $DIR/issue-77092.rs:13:26
+   |
+LL |         println!("{:?}", take_array_from_mut(&mut arr, i));
+   |                          ^^^^^^^^^^^^^^^^^^^ cannot infer the value of the constant `{_: usize}`
```

Comment:
Would it make more sense here to use a similar message to below?
E.g.
```
cannot infer the value of const parameter `N`
```

Reply:
the failed inference variable has a `ConstInference` origin, as it was created from unifying two uninferred consts vids.

So we don't have the info to emit a better error message in this case. We probably should "just" use one of the `ConstParam`s as the origin in that case. I think the reason we don't do so with types is due to variance.

Not completely sure about all of this though, so I would prefer to look into this in a followup PR